### PR TITLE
Fix the timescaledb-bitnami docker after the recent changes

### LINKED ISSUE #203: bitnami image fails at initdb
## Description

Since tag `2.9.0-pgX-bitnami`, timescaledb initialisation no longer works with just the `POSTGRES_PASSWORD` or `POSTGRES_PASSWORD_FILE` argument. This is problematic, because several helm charts (including the [`bitnami/postgres` version 12.1.19 helm chart](https://artifacthub.io/packages/helm/bitnami/postgresql)) assume that this is the only password variable needed.

## How to reproduce

The following does not work for versions 2.9.0 and 2.9.1:
```
docker run --rm -d -e POSTGRES_PASSWORD=test --name timescaledb timescale/timescaledb:2.9.0-pg12-bitnami 
```
with logs
```
$ docker logs -f timescaledb
postgresql 15:08:57.45 
postgresql 15:08:57.48 Welcome to the Bitnami postgresql container
postgresql 15:08:57.51 Subscribe to project updates by watching https://github.com/bitnami/containers
postgresql 15:08:57.54 Submit issues and feature requests at https://github.com/bitnami/containers/issues
postgresql 15:08:57.56 
postgresql 15:08:57.79 INFO  ==> ** Starting PostgreSQL setup **
postgresql 15:08:57.99 INFO  ==> Validating settings in POSTGRESQL_* env vars..
postgresql 15:08:58.11 INFO  ==> Loading custom pre-init scripts...
postgresql 15:08:58.19 INFO  ==> Initializing PostgreSQL database...
postgresql 15:08:58.54 INFO  ==> pg_hba.conf file not detected. Generating it...
postgresql 15:08:58.57 INFO  ==> Generating local authentication configuration
postgresql 15:09:03.19 INFO  ==> Starting PostgreSQL in background...
postgresql 15:09:03.74 INFO  ==> Changing password of postgres
postgresql 15:09:03.96 INFO  ==> Configuring replication parameters
postgresql 15:09:04.42 INFO  ==> Configuring synchronous_replication
postgresql 15:09:04.45 INFO  ==> Configuring fsync
postgresql 15:09:04.78 INFO  ==> Stopping PostgreSQL...
waiting for server to shut down.... done
server stopped
postgresql 15:09:04.96 INFO  ==> Loading custom scripts...
postgresql 15:09:05.03 INFO  ==> Loading user's custom files from /docker-entrypoint-initdb.d ...
postgresql 15:09:05.05 INFO  ==> Starting PostgreSQL in background...
Password for user postgres: 
psql: error: FATAL:  password authentication failed for user "postgres"
postgresql 15:09:05.64 INFO  ==> Stopping PostgreSQL...
waiting for server to shut down.... done
server stopped
```

and it works for version 2.8.1 and earlier
```
docker run --rm -d -e POSTGRES_PASSWORD=test timescale/timescaledb:2.8.1-pg12-bitnami 
```

Using `POSTGRESQL_PASSWORD` and/or `BITNAMI_APP_NAME` environment variables gives the same results.

### PR DESCRIPTION
**Pull request by akuzm**
They changed two things:
- disabled trust authentication for localhost
- started replacing the original config, overwriting shared_preload_libraries configured by us from their config environment variable POSTGRESQL_SHARED_PRELOAD_LIBRARIES.

This commit accounts for these things, and also makes some cosmetic changes.

Fixes #203

### CODE REVIEW

File: bitnami/Dockerfile [hunk 1]
```diff
@@ -19,12 +19,18 @@ RUN apk update && apk add --no-cache git gcc \
...
+# There are three types of SQL files, initialization, upgrade, and downgrade,
+# which we have to count separately, but it's hard to match with globs, and
+# there are also many upgrade/downgrade files per version, so just keep more of
+# them.
 USER 0
-RUN rm -f $(pg_config --sharedir)/extension/timescaledb*mock*.sql \
-    && if [ -f $(pg_config --pkglibdir)/timescaledb-tsl-1*.so ]; then rm -f $(ls -1 $(pg_config --pkglibdir)/timescaledb-tsl-1*.so | head -n -5); fi \
-    && if [ -f $(pg_config --pkglibdir)/timescaledb-1*.so ]; then rm -f $(ls -1 $(pg_config --pkglibdir)/timescaledb-*.so | head -n -5); fi \
-    && if [ -f $(pg_config --sharedir)/extension/timescaledb--1*.sql ]; then rm -f $(ls -1 $(pg_config --sharedir)/extension/timescaledb--1*.sql | head -n -5); fi
+RUN rm -vf $(pg_config --sharedir)/extension/timescaledb*mock*.sql \
```
Comment (akuzm) [inline bitnami/Dockerfile:28]: `[ -f ]` doesn't work with wildcards, so these checks were always erroring out.

File: bitnami/Dockerfile [hunk 2]
```diff
@@ -79,10 +86,13 @@ RUN set -ex \
             cmake \
     && apt-get clean -y \
     && rm -rf \
-      "${HOME}/.cache" \
+        /build \
```
Comment (akuzm) [inline bitnami/Dockerfile:89]: I guess we don't need the build artifacts or sources in the container.

File: bitnami/timescaledb-bitnami-entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+# We have to use the bitnami configuration variable to add timescaledb to
+# shared preload list, or else it gets overwritten.
+if [ -z "$POSTGRESQL_SHARED_PRELOAD_LIBRARIES" ]
+then
+    POSTGRESQL_SHARED_PRELOAD_LIBRARIES=timescaledb
+else
+    POSTGRESQL_SHARED_PRELOAD_LIBRARIES="$POSTGRESQL_SHARED_PRELOAD_LIBRARIES,timescaledb"
+fi
+export POSTGRESQL_SHARED_PRELOAD_LIBRARIES
```
Comment (kgyrtkirk) [inline bitnami/timescaledb-bitnami-entrypoint.sh:11]: I haven't seen this happening - I was only executing some basic commands;
but I've put together some basic test framework to run basic tests on the created images - I plan to contribute the tests to your branch (they don't succeed on the current main branch)
we could probably extend the test commands to cover for this as well?

File: .github/workflows/smoke-test.yml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  PLATFORM: linux/amd64
+
+jobs:
+  smoketest:
+    name: PG${{ matrix.pg }}-${{ matrix.type }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        pg: [14]
```
Comment (svenklemm) [inline .github/workflows/smoke-test.yml:20]: i think we wanna test all the versions we build here 
```suggestion
        pg: [12,13,14,15]
```

**Comment by kgyrtkirk** [comment]
I guess we were exporing testing of this thing in parallel - and I've came to a different solution implementing - its kinda built to be able to run more commands/etc

not sure if we want to have that or not; but I'll leave a link to it:
https://github.com/kgyrtkirk/timescaledb-docker/pull/1/files#diff-4ff2596354382ca8fa5c77e3623f0014d24963f7f49b0cfe9d3982f30a6142e2

**Review by svenklemm — APPROVED** [review]

**Comment by akuzm** [comment]
> I guess we were exporing testing of this thing in parallel - and I've came to a different solution implementing - its kinda built to be able to run more commands/etc
>
> not sure if we want to have that or not; but I'll leave a link to it:
https://github.com/kgyrtkirk/timescaledb-docker/pull/1/files#diff-4ff2596354382ca8fa5c77e3623f0014d24963f7f49b0cfe9d3982f30a6142e2

Maybe at some point we'll need to improve testing for this repository, right. Sven also mentioned that we should add tests with different architectures. For now, I'm going to merge this PR with a minimal test just to roll out the fix soon.

File: bitnami/timescaledb-bitnami-entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+# Fall through to the original entrypoint.
+/opt/bitnami/scripts/postgresql/entrypoint.sh "$@"
```
Comment (blootsvoets) [inline bitnami/timescaledb-bitnami-entrypoint.sh:14]: Should this be an `exec` instead?

Comment (akuzm) [reply bitnami/timescaledb-bitnami-entrypoint.sh:14]: Yes, you're right, this wrapper script shouldn't change PID 1. I'll fix it.

Comment (akuzm) [reply bitnami/timescaledb-bitnami-entrypoint.sh:14]: https://github.com/timescale/timescaledb-docker/pull/206

**Pull request merged**